x-pack/plugins/licensing/server/plugin.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Observable } from 'rxjs';
+import { first, map } from 'rxjs/operators';
+import moment from 'moment';
+import {
+  CoreSetup,
+  CoreStart,
+  Logger,
+  Plugin as CorePlugin,
+  PluginInitializerContext,
+} from 'src/core/server';
+import { Poller } from '../../../../src/core/utils/poller';
+import { LicensingConfigType, LicensingPluginSetup, ILicense } from './types';
+import { LicensingConfig } from './licensing_config';
+import { License } from './license';
+
+export class Plugin implements CorePlugin<LicensingPluginSetup> {
+  private readonly logger: Logger;
+  private readonly config$: Observable<LicensingConfig>;
+  private poller!: Poller<ILicense>;
+
+  constructor(private readonly context: PluginInitializerContext) {
+    this.logger = this.context.logger.get();
+    this.config$ = this.context.config
+      .create<LicensingConfigType | { config: LicensingConfigType }>()
+      .pipe(
+        map(config =>
+          'config' in config
+            ? new LicensingConfig(config.config, this.context.env)
+            : new LicensingConfig(config, this.context.env)
+        )
+      );
+  }
+
+  private hasLicenseInfoChanged(newLicense: any) {
```

Comment:
can we use something like `lodash.isEqual` or Rx.distinctUntilChanged? and with `any` it's hard to say that all fields are covered by https://github.com/elastic/kibana/pull/43623/files#diff-6cafae74a549dc892111f589f7faa6a7R48-R50

Reply:
This is custom logic for the previous x-pack changing implementation, so we don't want to deviate in case there are special edge cases.